ironmon_tracker/Program.lua

```diff
@@ -263,6 +284,13 @@ function Program.updatePokemonTeams()
 				-- newPokemonData.trainerID = nil
 
 				Tracker.addUpdatePokemon(newPokemonData, personality, true)
+
+				-- If this is a newly caught Pokémon, track all of its moves. Can't do this later cause TMs/HMs
+				if previousPersonality == 0 then
+					for _, move in ipairs(newPokemonData.moves) do
+						Tracker.TrackMove(newPokemonData.pokemonID, move.id, newPokemonData.level)
+					end
+				end
```

Comment:
You are correct. I should have noted this in the PR.

I thought about this for a while, and while yes it does mean that catching a mon or starting with a starter in the lab (basically receiving a mon into your party for the first time) will tracks those four moves, I think this is somewhat needed to help fill out the tracked move history page. It's more likely a player is going to be catching mons as pivots and wanting to use them, than they are going to catch a mon and release it due to some circumstances.

There wasn't a quick and easy solution for me to get around this. The logic that is required to determine what moves to track and when not to track them is wide.

For example
- Catch & Release: dont track
- Catch & Pivot, but then catch another and release the first: track or dont track?
- Catch & use: track or dont track?
   - What if they taught it the TM from the starter

For the time being i'd like to try it as this, unless you think its more likely to be detrimental to most players gameplay experiences.

Reply:
yeah that makes sense, i agree it's a rather niche scenario especially since most of the scenarios someone would catch & release without looking is in early game when scouting pivots so those moves would be forgotten 99% of the time anyways (and if not then most likely would be seen in battle) so it's not a huge detriment. As long as it's noted that it does behave like this